### PR TITLE
Enable Downloading Large Files

### PR DESCRIPTION
**Pull request by wistuba**
Current code loads downloaded file into memory first before writing to disk. This does not work for large files.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/renate/utils/file.py [hunk 1]
```diff
@@ -305,8 +305,11 @@ def download_file(
     if src_bucket is None:
         if not os.path.exists(os.path.join(data_path, dataset_name)):
             os.makedirs(os.path.join(data_path, dataset_name))
-        response = requests.get(os.path.join(url, file_name), allow_redirects=True)
-        open(os.path.join(data_path, dataset_name, file_name), "wb").write(response.content)
+        with requests.get(os.path.join(url, file_name), allow_redirects=True, stream=True) as r:
+            r.raise_for_status()
+            with open(os.path.join(data_path, dataset_name, file_name), "wb") as f:
+                for chunk in r.iter_content(chunk_size=8192):
```
Comment (prabhuteja12) [inline src/renate/utils/file.py:311]: Is the 8192 number of significance?

**Review by wesk — COMMENTED** [review]
What happens if it fails? Is there a try/except somewhere else?

File: src/renate/utils/file.py (hunk 1 above)
Comment (wistuba) [reply src/renate/utils/file.py:311]: No, we could also use 1024. Default is 1: https://requests.readthedocs.io/en/latest/api/#requests.Response.iter_content

**Comment by wistuba** [comment]
> What happens if it fails? Is there a try/except somewhere else?

If it fails, it fails with one of these exceptions: https://requests.readthedocs.io/en/latest/user/quickstart/#errors-and-exceptions
What do you recommend doing in the `except` part? If we cannot download the file, we cannot continue.

**Review by prabhuteja12 — APPROVED** [review]

**Comment by wesk** [comment]
You could set up automatic retries to handle transient failures: https://requests.readthedocs.io/en/latest/user/advanced/?highlight=retry#example-automatic-retries

But agree code is good to start with, and you can add later if it's ever a problem.

**Pull request merged**